### ISSUE TITLE
3.1.0 release cause a "TERRAFORM CRASH" with ECR resource

### ISSUE THREAD

**Issue by yogeek** [terraform-providers/terraform-provider-aws#14536]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
terraform -v                                            
Terraform v0.12.29
+ provider.aws v3.1.0
```

### Affected Resource(s)

* aws_ecr_repository

### Terraform Configuration Files

ecr.tf
```hcl
resource "aws_ecr_repository" "name" {

    for_each             = toset(var.images_list)
    name                 = each.value

    image_tag_mutability = "MUTABLE"
    image_scanning_configuration {
        scan_on_push = true
    }
    tags = merge(local.tags,
         map(
            "Repository_Url",var.tag_repository_url,
            "Commit_ID",var.tag_commit_id
         )
    )
}
```

variables.tf
```
[...]
variable "images_list" {
  description = "Repositories to create."
  default     = []
}
[...]
```

images.tfvars
```
images_list = [
  "image1",
  "image2",
  [...]
]

```

### Panic Output

[crash.log](https://gist.github.com/yogeek/0dc45be1c993aedbff189a3d3f2d2680)

### Expected Behavior

no panic error (at least an "explained" error ;) )

### Actual Behavior

```
terraform plan -var-file=images.tfvars'
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

aws_ecr_repository.name["image1"]: Refreshing state... [id=image1]
aws_ecr_repository.name["image2"]: Refreshing state... [id=image2]

Error: rpc error: code = Unavailable desc = transport is closing
Error: rpc error: code = Unavailable desc = transport is closing
Error: rpc error: code = Canceled desc = context canceled
[...]
2020-08-10T11:34:12.644+0200 [DEBUG] plugin: plugin exited

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

SECURITY WARNING: the "crash.log" file that was created may contain 
sensitive information that must be redacted before it is safe to share 
on the issue tracker.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Error: rpc error: code = Canceled desc = context canceled

```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

no error when using the previous provider version :

```
provider "aws" {
  region = var.aws_region
  version = "<= 3.0.0"
}
```

**Comment by ewbankkit** [comment]
Relevant part of the crash log:

```
2020-08-10T11:34:12.634+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: panic: runtime error: invalid memory address or nil pointer dereference
2020-08-10T11:34:12.634+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x3c62916]
2020-08-10T11:34:12.634+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: 
2020-08-10T11:34:12.634+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: goroutine 22 [running]:
2020-08-10T11:34:12.634+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: github.com/terraform-providers/terraform-provider-aws/aws.flattenEcrRepositoryEncryptionConfiguration(...)
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_ecr_repository.go:234
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: github.com/terraform-providers/terraform-provider-aws/aws.resourceAwsEcrRepositoryRead(0xc000120770, 0x4e82720, 0xc0002b0f00, 0xc000120770, 0x0)
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/aws/resource_aws_ecr_repository.go:195 +0xa56
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc0006cf320, 0xc0017f0460, 0x4e82720, 0xc0002b0f00, 0xc00065d148, 0x0, 0x0)
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform-plugin-sdk/helper/schema/resource.go:460 +0x119
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadResource(0xc000288328, 0x64bbc20, 0xc0017fa030, 0xc000cf42a0, 0xc000288328, 0x0, 0xc00177cba0)
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin/grpc_provider.go:525 +0x3d8
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadResource_Handler(0x588db40, 0xc000288328, 0x64bbc20, 0xc0017fa030, 0xc000cf4240, 0x0, 0x64bbc20, 0xc0017fa030, 0xc0017fc000, 0x31a)
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5/tfplugin5.pb.go:3269 +0x217
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: google.golang.org/grpc.(*Server).processUnaryRPC(0xc00033f500, 0x64dce40, 0xc00169a900, 0xc0017f8000, 0xc00090ae10, 0x97f0d90, 0x0, 0x0, 0x0)
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:1171 +0x50a
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: google.golang.org/grpc.(*Server).handleStream(0xc00033f500, 0x64dce40, 0xc00169a900, 0xc0017f8000, 0x0)
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:1494 +0xccd
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0007e2840, 0xc00033f500, 0x64dce40, 0xc00169a900, 0xc0017f8000)
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:834 +0xa1
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: created by google.golang.org/grpc.(*Server).serveStreams.func1
2020-08-10T11:34:12.635+0200 [DEBUG] plugin.terraform-provider-aws_v3.1.0_x5: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-aws/vendor/google.golang.org/grpc/server.go:832 +0x204
```

**Comment by anGie44** [comment]
The fix for this has been merged and will release with `v3.2.0` of the Terraform AWS Provider.
